pkg/uid/uid.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+import (
+	"errors"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// A UidRange structure used to set uidshift and its range.
+type UidRange struct {
+	UidShift uint64
+	UidCount uint64
+}
+
+func NewUidRange(uidshift uint64, uidcount uint64) *UidRange {
+	ur := &UidRange{
+		uidshift,
+		uidcount,
+	}
+	return ur
+}
+
+func GenerateContainerID(containerid uint64) uint64 {
+	var containerID uint64 = containerid
+
+	if containerID < 0x00010000 {
+		rand.Seed(time.Now().UnixNano())
+		containerID = uint64(rand.Int63n(0xFFFDFFFF)) + 0X00010000
+	}
+
+	return containerID
+}
+
+func GenerateUidShift(containerid uint64) uint64 {
+	var uidShift uint64
+	rand.Seed(time.Now().UnixNano())
+	containerUid := rand.Intn(0x0000FFFF)
+
+	uidShift = GenerateContainerID(containerid) | uint64(containerUid)
+
+	return uidShift
+}
+
+func GenerateUidRange(containerid uint64, uidcount uint64) *UidRange {
+	uidShift := GenerateUidShift(containerid)
+	ur := NewUidRange(uidShift, uidcount)
+	return ur
+}
+
+func SetUidRange(uidrange *UidRange, containerid uint64, uidcount uint64) {
```

Comment:
Alban it's in case we want to pass specific containerIDs. I'm sure we will use it later where multiple containers must be sync'ed using the db locking mechanism or any other solution.